app/src/main/java/com/example/srv_twry/studentcompanion/POJOs/FlashCard.java

```diff
@@ -31,24 +47,108 @@ public String getAnswer() {
         return answer;
     }
 
+    public int getContentID() {
+        return contentID;
+    }
+
+
     @Override
     public int describeContents() {
         return 0;
     }
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
+        dest.writeInt(this.contentID);
         dest.writeString(this.topicName);
         dest.writeString(this.question);
         dest.writeString(this.answer);
     }
 
     private FlashCard(Parcel in) {
+        contentID = in.readInt();
         this.topicName = in.readString();
         this.question = in.readString();
         this.answer = in.readString();
     }
 
+    public void deleteFromDB(final Activity parentContext, final boolean closeParent){
+        //Show the alert dialog
+        AlertDialog.Builder alertDialog = new AlertDialog.Builder(parentContext);
+        alertDialog.setTitle(R.string.confirm_delete);
+        alertDialog.setMessage(R.string.are_you_sure_you_want_to_delete_this_card);
+        alertDialog.setIcon(R.drawable.ic_delete_black);
+
+        alertDialog.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog,int which) {
+                String idString = Integer.toString(contentID);
+                Uri deleteFlashCardIndividual = DatabaseContract.FlashCardsEntry.CONTENT_URI_FLASH_CARDS.buildUpon().appendPath(idString).build();
+                int itemsDeleted = parentContext.getContentResolver().delete(deleteFlashCardIndividual,null,null);
+
+                if (itemsDeleted >0 ){
+                    Toast.makeText(parentContext, R.string.card_deleted_successfully,Toast.LENGTH_SHORT).show();
+                }else{
+                    Toast.makeText(parentContext, R.string.cannot_delete_card, Toast.LENGTH_SHORT).show();
+                }
+                if(closeParent){
+                    parentContext.finish();
+                }
+            }
+        });
+
+        alertDialog.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+
+        alertDialog.show();
+    }
+
+    public static FlashCard getFlashCardFromCursor(Cursor mCursor){
+        String tag = "getFlashcardfromcursor";
```

Comment:
I would suggest using standard Java convention of naming constants by all capitals.